internal/provider/resource_assigned_managed_identity.go

```diff
@@ -0,0 +1,301 @@
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"workspace_id": schema.StringAttribute{
+				MarkdownDescription: "The ID of the assigned-to workspace.",
+				Description:         "The ID of the assigned-to workspace.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+	}
+}
+
+// Configure lets the provider implement the ResourceWithConfigure interface.
+func (t *assignedManagedIdentityResource) Configure(_ context.Context,
+	req resource.ConfigureRequest, _ *resource.ConfigureResponse,
+) {
+	if req.ProviderData == nil {
+		return
+	}
+	t.client = req.ProviderData.(*tharsis.Client)
+}
+
+func (t *assignedManagedIdentityResource) Create(ctx context.Context,
+	req resource.CreateRequest, resp *resource.CreateResponse,
+) {
+	// Retrieve values from assigned managed identity.
+	var assignment AssignedManagedIdentityModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &assignment)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get the workspace in order to have the path.
+	workspace, err := t.client.Workspaces.GetWorkspace(ctx,
+		&ttypes.GetWorkspaceInput{
+			ID: ptr.String(assignment.WorkspaceID.ValueString()),
+		},
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error getting workspace",
+			err.Error(),
+		)
+		return
+	}
+
+	// Create the assigned managed identity. (In other words, assign the managed identity to the workspace.)
+	managedIdentityID := assignment.ManagedIdentityID.ValueString()
+	_, err = t.client.ManagedIdentity.AssignManagedIdentityToWorkspace(ctx,
+		&ttypes.AssignManagedIdentityInput{
+			ManagedIdentityID: &managedIdentityID,
+			WorkspacePath:     workspace.FullPath,
+		})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating assigned managed identity",
+			err.Error(),
+		)
+		return
+	}
+
+	created := AssignedManagedIdentityModel{
+		ID:                types.StringValue(uuid.New().String()), // computed with no input from any other resource
+		ManagedIdentityID: types.StringValue(managedIdentityID),
+		WorkspaceID:       types.StringValue(workspace.Metadata.ID),
+	}
+
+	// Map the response body to the schema.  (There are no computed values to update to the plan.)
+	t.copyAssignedManagedIdentity(&created, &assignment)
+
+	// Set the response state to the fully-populated plan, whether or not there is an error.
+	resp.Diagnostics.Append(resp.State.Set(ctx, assignment)...)
+}
+
+func (t *assignedManagedIdentityResource) Read(ctx context.Context,
+	req resource.ReadRequest, resp *resource.ReadResponse,
+) {
+	// Get the current state.
+	var state AssignedManagedIdentityModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get the workspace from Tharsis so we can have its ID.
+	workspace, err := t.client.Workspaces.GetWorkspace(ctx,
```

Comment:
Any reason the workspace is being queried here?

Reply:
The workspace ID is needed in line 200 to create a from-Tharsis model.  I have the workspace ID from the state, but not from Tharsis.  In the Read method, are we guaranteed that the state is complete and accurate?  If so, that query could be removed and the from-state workspace ID used instead of getting it from Tharsis.

Reply:
It should be using the workspace ID from the state, if that isn't the same ID then something is wrong, and the provider should rightfully error.